source/parser.hera

```diff
@@ -608,6 +605,19 @@ CallExpression
 
     return member
 
+# NOTE: Special `return.value` (and `return =` shorthand)
+# for changing the automatic return value of function
+ReturnValue
+  ( "return.value" NonIdContinue ) / ( Return &AfterReturnShorthand ) ->
+    return { type: "ReturnValue", children: [$1[0]] }
+
+AfterReturnShorthand
+  WAssignmentOp
+  UpdateExpressionSymbol
+  __ Colon
```

Comment:
This could pick up a `Label` on the following line, we may want to disallow `__` and enforce same line or use a more specific rule.